r/R/zzz.R

```diff
@@ -0,0 +1,88 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# nocov start
+.onLoad <- function(...) {
+  s3_register("arrow::infer_type", "nanoarrow_array")
+  s3_register("arrow::as_data_type", "nanoarrow_schema")
+  s3_register("arrow::as_schema", "nanoarrow_schema")
+  s3_register("arrow::as_arrow_array", "nanoarrow_array")
+  s3_register("arrow::as_chunked_array", "nanoarrow_array")
+  s3_register("arrow::as_record_batch", "nanoarrow_array")
+  s3_register("arrow::as_arrow_table", "nanoarrow_array")
+  s3_register("arrow::as_record_batch_reader", "nanoarrow_array_stream")
+}
+
+s3_register <- function(generic, class, method = NULL) {
```

Comment:
This is copied from another package, yeah? Should provide attribution.

Reply:
This may require a note in a LICENSE file, as we do in apache/arrow. Fortunately it is [permissively licensed (MIT)](https://github.com/r-lib/vctrs/blob/main/DESCRIPTION#L26).

Reply:
Oh interesting